library/Zend/Stdlib/CallbackHandler.php

```diff
@@ -103,34 +100,22 @@ public function call(array $args = array())
             }
         }
 
-        // Minor performance tweak; use call_user_func() until > 3 arguments
+        // Minor performance tweak; use call_user_func_array() with > 3 arguments
         // reached
         switch ($argCount) {
             case 0:
-                if (static::$isPhp54) {
-                    return $callback();
-                }
-                return call_user_func($callback);
+                return $callback();
             case 1:
-                if (static::$isPhp54) {
-                    return $callback(array_shift($args));
-                }
-                return call_user_func($callback, array_shift($args));
+                return $callback(array_shift($args));
             case 2:
                 $arg1 = array_shift($args);
                 $arg2 = array_shift($args);
-                if (static::$isPhp54) {
-                    return $callback($arg1, $arg2);
-                }
-                return call_user_func($callback, $arg1, $arg2);
+                return $callback($arg1, $arg2);
             case 3:
                 $arg1 = array_shift($args);
                 $arg2 = array_shift($args);
                 $arg3 = array_shift($args);
-                if (static::$isPhp54) {
-                    return $callback($arg1, $arg2, $arg3);
-                }
-                return call_user_func($callback, $arg1, $arg2, $arg3);
+                return $callback($arg1, $arg2, $arg3);
             default:
                 return call_user_func_array($callback, $args);
```

Comment:
well, that would bring the whole method back to its state before 6f18994c0f528130c0963063a37db27767590dec

i got ridden of the shifts, it should be the right compromise now